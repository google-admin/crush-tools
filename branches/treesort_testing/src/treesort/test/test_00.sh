test_number=00
description="one column sort"

subtest_desc=("first column"
							"second column using index"
							"first column reversed"
							"second column reversed with index")
subtest_opts=(
'-K AA'
'-k 2 -p'
'-K ^AA'
'-k ^2 -p'
)

for subtest in `seq 0 3`; do
  expected="$test_dir/test_$test_number.$subtest.expected"
  outfile="$test_dir/test_$test_number.$subtest.actual"

  $bin ${subtest_opts[$subtest]} \
       "$test_dir/test.in" \
       > "$outfile"

  if [ $? -ne 0 ] ||
     [ "`diff -q $outfile $expected`" ]; then
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" FAIL
  else
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" PASS
    rm "$outfile"
  fi
done
