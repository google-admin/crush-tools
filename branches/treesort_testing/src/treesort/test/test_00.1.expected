BBþAAþCCþDD
0dGw4UOK0Pþ279þ2009-10-26þ17.0268
3XbaqX1keSþ843þ2009-10-02þ32.4092
49Uj8LVH5jþ222þ2009-10-02þ13.5003
4j4n56w__lþ577þ2009-10-25þ14.3872
5TcvIxjquUþ577þ2009-10-15þ32.4092
6ff-bhImN7þ918þ2009-10-10þ26.6454
7YaqxPKpCRþ740þ2009-10-01þ17.0268
A-cSNvcqnzþ468þ2009-10-01þ28.4036
ENItEY_hIBþ991þ2009-10-23þ14.9256
OCS81RVo5jþ222þ2009-10-28þ32.4092
OcNUM48IS6þ740þ2009-10-15þ22.8364
W2OIToxqAMþ843þ2009-10-04þ17.0268
WDEGX5smiWþ279þ2009-10-15þ20.7444
ZfcHogunHrþ843þ2009-10-01þ28.4036
afUGgY6q4jþ279þ2009-10-14þ20.7444
arzhFJlrMBþ577þ2009-10-14þ28.4036
tHGNJyQcu1þ278þ2009-10-01þ32.4092
vOXGmYLegsþ468þ2009-10-04þ29.5253
xVciPqOBiQþ965þ2009-10-14þ17.0268
z3cqeuch4gþ918þ2009-10-28þ20.7444
